File: output/CodeGen/codenet/C++/Java/s568362035.java
public class s568362035 {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        int m = in.nextInt();
        int Q = in.nextInt();
        int[] a = new int[Q + 1];
        for (int i = 1; i <= Q; i++) {
            a[i] = in.nextInt();
        }
        int[] b = new int[m];
        boolean[] vis = new boolean[n + 1];
        int[] cnt = new int[n + 1];
        for (int i = 0; i < m; i++) {
            b[i] = in.nextInt();
            vis[b[i]] = true;
        }
        for (int i = m - 1; i >= 0; i--) {
            if (!vis[b[i]]) {
                b[i] = in.nextInt();
                vis[b[i]] = true;
            }
        }
        for (int i = 1; i <= m; i++) {
            int t = b[i];
            if (cnt[t - 1] > 0) {
                --cnt[t - 1];
                ++cnt[t];
                vis[t] = true;
            } else if (!vis[t]) {
                System.out.println("No");
                return;
            }
        }
        boolean flag = true;
        int l;
        for (l = 0; cnt[l] == 0; ++l);
        for (int i = l + 1; i < m; ++i) {
            flag &= b[i] < b[i + 1];
        }
        System.out.println(flag? "Yes" : "No");
    }
}

File: output/CodeGen/codenet/C++/Python/s568362035.py
def main():
    n, m, Q = map(int, raw_input().split())
    a = map(int, raw_input().split())
    b = []
    id = []
    vis = [False] * (n + 1)
    cnt = [0] * (n + 1)
    for i in range(m):
        b.append(int(raw_input()))
        id.append(i)
    for i in range(m, n + 1):
        if not vis[int(raw_input())]:
            b.append(i)
            vis[i] = True
            id.append(i)
    for i in range(m):
        if not vis[i]:
            b.append(i)
            vis[i] = True
            id.append(i)
    for i in range(m, n + 1):
        if not vis[i]:
            b.append(i)
            vis[i] = True
            id.append(i)
    for i in range(m):
        cnt[b[i]] += 1
    for i in range(m, n + 1):
        cnt[b[i]] -= 1
        if cnt[b[i]] == 0:
            cnt[b[i - 1]] += 1
            vis[b[i - 1]] = True
            id[b[i - 1]] = i - 1
    for i in range(m):
        if cnt[b[i]] == 0:
            cnt[b[i + 1]] += 1
            vis[b[i + 1]] = True
            id[b[i + 1]] = i + 1
    flag = True
    l = 0
    for i in range(n):
        if cnt[l] == 0:
            l += 1
    for i in range(l, m):
        flag &= b[i] < b[i + 1]
    if flag:
        print "Yes"
    else:
        print "No"

main()